web/views/student.py

```diff
@@ -407,7 +404,7 @@ def task_detail(request, assignment_id, submit_num=None, login=None):
         )
 
     if current_submit:
-        data = {**data, **get(current_submit)}
+        data = {**data, **get(current_submit).__dict__}
```

Comment:
:D
```suggestion
        data = {**data, "submit": current_submit, "results":  get_submit_data(current_submit).results}
```

Reply:
Oh, sure, having the `data` object be a dataclass would indeed be nice. That's for a separate PR though, and it's a much larger refactoring :)